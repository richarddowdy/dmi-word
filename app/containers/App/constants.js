/*
 * AppConstants
 */

export const ADD_STRINGS_SUCCESS = 'word-dmi/Form/ADD_STRINGS_SUCCESS';
export const ADD_STRINGS_FAIL = 'word-dmi/Form/ADD_STRINGS_FAIL';
