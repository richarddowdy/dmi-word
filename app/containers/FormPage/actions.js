/*
 *
 * Form actions
 *
 */

import {
  CHANGE_INPUT,
  ADD_NEW_STRING,
  LOAD_STRINGS_SUCCESS,
  ADD_STRINGS_SUCCESS,
  ADD_STRINGS_FAIL,
} from './constants';

export function changeInput(newString) {
  return {
    type: CHANGE_INPUT,
    newString,
  };
}

export function addNewString(newString) {
  return {
    type: ADD_NEW_STRING,
    newString,
  };
}

export function stringAdd(stringsArray) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    stringsArray,
  };
}

export function stringFailed(error) {
  return {
    type: ADD_STRINGS_FAIL,
    error,
  };
}

export function stringSuccess(success) {
  return {
    type: ADD_STRINGS_SUCCESS,
    success,
  };
}
