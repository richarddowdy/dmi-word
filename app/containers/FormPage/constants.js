/*
 *
 * Form constants
 *
 */

export const CHANGE_INPUT = 'word-dmi/Form/CHANGE_INPUT';
export const ADD_NEW_STRING = 'word-dmi/Form/ADD_NEW_STRING';
export const LOAD_STRINGS_SUCCESS = 'word-dmi/Form/LOAD_STRINGS_SUCCESS';
