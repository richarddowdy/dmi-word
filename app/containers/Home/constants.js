/*
 *
 * Home constants
 *
 */

export const DEFAULT_ACTION = 'app/Home/DEFAULT_ACTION';
export const REQUEST_API_STRINGS = 'word-dmi/Home/REQUEST_API_STRINGS';
export const RECEIVE_API_STRINGS = 'word-dmi/Home/RECEIVE_API_STRINGS';
export const LOAD_STRINGS_SUCCESS = 'word-dmi/Home/LOAD_STRINGS_SUCCESS';
export const DELETE_STRING = 'word-dmi/Home/DELETE_STRING';
export const DELETE_SUCCESS = 'word-dmi/Home/DELETE_SUCCESS';
export const DELETE_FAIL = 'word-dmi/Home/DELETE_FAIL';
export const RESET_SUCCESS = 'word-dmi/Home/RESET_SUCCESS';
