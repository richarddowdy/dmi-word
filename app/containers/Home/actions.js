/*
 *
 * Home actions
 *
 */

import { DEFAULT_ACTION, REQUEST_API_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export function loadStrings() {
  return {
    type: REQUEST_API_STRINGS,
  };
}

export function stringsLoaded(strings) {
  console.log("pushing to redux", strings)
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}
