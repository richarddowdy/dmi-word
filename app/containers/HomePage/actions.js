/*
 *
 * Home actions
 *
 */

import {
  REQUEST_API_STRINGS,
  LOAD_STRINGS_SUCCESS,
  DELETE_STRING,
  DELETE_FAIL,
  DELETE_SUCCESS,
} from './constants';

import { LOAD_STRINGS_FAIL } from '../FormPage/constants';

export function loadStrings() {
  return {
    type: REQUEST_API_STRINGS,
  };
}

export function stringsLoaded(strings) {
  return {
    type: LOAD_STRINGS_SUCCESS,
    strings,
  };
}

export function stringsFailed(error) {
  return {
    type: LOAD_STRINGS_FAIL,
    error,
  };
}

export function deleteString(string) {
  return {
    type: DELETE_STRING,
    string,
  };
}

export function deleteFailed(error) {
  return {
    type: DELETE_FAIL,
    error,
  };
}

export function deleteSuccess(strings) {
  return {
    type: DELETE_SUCCESS,
    strings,
  };
}
