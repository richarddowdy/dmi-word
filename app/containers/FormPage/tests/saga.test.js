/**
 * Test sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { take, call, put, select, takeLatest } from 'redux-saga/effects';
// import formSaga from '../saga';

import { ADD_NEW_STRING, LOAD_STRINGS_SUCCESS } from '../constants';
import { addNewString, stringAdd, stringFailed, stringSuccess } from '../actions';

import formSaga, { addString } from '../saga';

// const generator = formSaga();

const string = ['testing'];

describe('formSaga Saga', () => {
  let addStringGenerator;

  beforeEach(() => {
    addStringGenerator = addString();

    const selectDescriptor = addStringGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = addStringGenerator.next(string).value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the loadStringsSuccess action if the request is successfull', () => {
    const newString = 'testing';
    const response = ['testing'];
    let putDescriptor = addStringGenerator.next().value;
    console.log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!", putDescriptor);
    expect(putDescriptor).toEqual(put(stringAdd(string)));
  });

});

describe('form saga', () => {
  const testFormSaga = formSaga();

  it('should watch for ADD_NEW_STRING action', () => {
    const takeLatestDescriptor = testFormSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(ADD_NEW_STRING, addString));
  });
})
