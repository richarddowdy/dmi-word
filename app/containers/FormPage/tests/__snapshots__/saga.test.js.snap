// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formSaga Saga Expect to have unit tests specified 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`formSaga Saga Expect to have unit tests specified 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`formSaga Saga should dispatch the loadStringsSuccess action if the request is successfull 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`formSaga Saga should dispatch the loadStringsSuccess action if the request is successfull 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;
